addons/website_sale/models/sale_order.py

```diff
@@ -353,13 +349,49 @@ def action_recovery_email_send(self):
                 'default_composition_mode': 'mass_mail' if len(self.ids) > 1 else 'comment',
                 'default_res_id': self.ids[0],
                 'default_model': 'sale.order',
-                'default_use_template': bool(template_id),
-                'default_template_id': template_id,
+                'default_use_template': bool(template),
+                'default_template_id': template.id if template else False,
                 'website_sale_send_recovery_email': True,
                 'active_ids': self.ids,
             },
         }
 
+    @api.multi
+    def _get_recovery_template(self):
+        """
+        Return the cart recovery template for a set of orders.
+        If they all belong to the same website, we return the default template;
+        otherwise we return the website-specific template.
+        """
+        try:
```

Comment:
What could actually raise here? There is `raise_if_not_found=False` on the first line, and the second line is just a `filter` and a `mapped`, that's not supposed to raise I believe. Need to double check why it was introduced in the first place.